### PR TITLE
Code refactoring for ember-routing package

### PR DESCRIPTION
**Pull request by artemgurzhii**

### CODE REVIEW

File: packages/ember-routing/lib/system/dsl.js [hunk 1]
```diff
@@ -133,7 +131,7 @@ function createRoute(dsl, name, options, callback) {
   dsl.push(options.path, fullName, callback, options.serialize);
 }
 
-DSL.map = function(callback) {
+DSL.map = callback => {
```
Comment (locks) [inline packages/ember-routing/lib/system/dsl.js]: I'm worried about this. cc @rwjblue

**Comment by artemgurzhii** [comment]
Can I try to do some optimization for this package? After all, I will show benchmark tests for each method that I have optimized.

File: packages/ember-routing/lib/ext/controller.js [hunk 2]
```diff
@@ -55,8 +55,8 @@ ControllerMixin.reopen({
     @method _qpChanged
     @private
   */
-  _qpChanged(controller, _prop) {
-    let prop = _prop.substr(0, _prop.length - 3);
+  _qpChanged(controller, prop) {
```
Comment (stefanpenner) [inline packages/ember-routing/lib/ext/controller.js]: we typically prefer to not overwrite argument bindings.

**Comment by stefanpenner** [comment]
@artemgurzhii this looks awesome, thank you!

**Pull request merged**